app/controllers/two_factor_auths_controller.rb

```diff
@@ -0,0 +1,55 @@
+class TwoFactorAuthsController < ApplicationController
+  before_action :redirect_to_root, unless: :signed_in?
+  before_action :require_no_auth, only: %i[new create]
+  before_action :require_auth_set, only: :destroy
+  helper_method :issuer
+
+  def new
+    @seed = ROTP::Base32.random_base32
+    session[:mfa_seed] = @seed
+    text = ROTP::TOTP.new(@seed, issuer: issuer).provisioning_uri(current_user.email)
+    @qrcode_svg = RQRCode::QRCode.new(text, level: :l).as_svg
+  end
+
+  def create
+    seed = session[:mfa_seed]
+    session[:mfa_seed] = nil
+    totp = ROTP::TOTP.new(seed, issuer: issuer)
+    if totp.verify(params[:otp])
+      current_user.enable_mfa!(seed, :auth_only)
+      flash[:success] = t('.enable_success')
+      render :recovery
+    else
+      flash[:error] = t('.otp_auth_failed')
+      redirect_to edit_profile_url
+    end
+  end
+
+  def destroy
+    if current_user.otp_verified?(params[:otp])
+      flash[:success] = t('.disable_success')
+      current_user.disable_mfa!
+    else
+      flash[:error] = t('profiles.mfa_enable.otp_auth_failed')
```

Comment:
I got `t('profiles.mfa_enable.otp_auth_failed')` as not found.

Reply:
oh! sorry, that should be `t('two_factor_auths.create.otp_auth_failed')`

Reply:
Is semantics of `flash[:notice]` more suitable here?

Reply:
error is fine.